test/controller/confirm.company.spec.unit.ts

```diff
@@ -105,6 +106,62 @@ describe("confirm company tests", () => {
     expect(response.status).toEqual(302);
     expect(response.header.location).toEqual(OBJECTIONS_ENTER_INFORMATION);
   });
+
+  it("should render the notice expired page when an api error occurs with the status INELIGIBLE_COMPANY_STRUCK_OFF", async () => {
+
+    const apiError: ApiError = {
+      data: {
+        status: "INELIGIBLE_COMPANY_STRUCK_OFF"
+      },
+      message: "There is an error",
+      status: 400,
+    };
+
+    mockGetObjectionSessionValue.mockReset();
+    mockGetObjectionSessionValue.mockImplementation(() => dummyCompanyProfile);
+
+    mockSetObjectionSessionValue.mockReset();
+
+    mockCreateNewObjection.mockReset();
+    mockCreateNewObjection.mockImplementation(() => {
+      throw apiError
+    });
+
+    const response = await request(app).post(OBJECTIONS_CONFIRM_COMPANY)
+      .set("Referer", "/")
+      .set("Cookie", [`${COOKIE_NAME}=123`]);
+
+    expect(response.status).toEqual(302);
+    expect(response.header.location).toEqual(OBJECTIONS_NOTICE_EXPIRED);
+  });
+
+  it("should render the no strike off page when an api error occurs with the status INELIGIBLE_NO_DISSOLUTION_ACTION", async () => {
+
+    const apiError: ApiError = {
+      data: {
+        status: "INELIGIBLE_NO_DISSOLUTION_ACTION"
+      },
+      message: "There is an error",
+      status: 400,
+    };
+
+    mockGetObjectionSessionValue.mockReset();
+    mockGetObjectionSessionValue.mockImplementation(() => dummyCompanyProfile);
+
+    mockSetObjectionSessionValue.mockReset();
+
+    mockCreateNewObjection.mockReset();
+    mockCreateNewObjection.mockImplementation(() => {
+      throw apiError
+    });
+
+    const response = await request(app).post(OBJECTIONS_CONFIRM_COMPANY)
+      .set("Referer", "/")
+      .set("Cookie", [`${COOKIE_NAME}=123`]);
+
+    expect(response.status).toEqual(302);
+    expect(response.header.location).toEqual(OBJECTIONS_NO_STRIKE_OFF);
+  });
```

Comment:
add a test for an unknown objection status (data.status) and it should show the generic error page